src/raft/replication.c

```diff
@@ -1140,25 +1208,38 @@ int replicationAppend(struct raft *r,
 
 	assert(r->state == RAFT_FOLLOWER);
 
+	request = raft_malloc(sizeof *request);
+	if (request == NULL) {
+		rv = RAFT_NOMEM;
+		goto err;
+	}
+	sm_init(&request->sm, af_invariant, NULL, af_states, "append-follower",
+		AF_START);
+
 	*rejected = args->prev_log_index;
 	*async = false;
 
 	/* Check the log matching property. */
 	match = checkLogMatchingProperty(r, args);
-	if (match != 0) {
-		assert(match == 1 || match == -1);
-		return match == 1 ? 0 : RAFT_SHUTDOWN;
+	if (match == 1) {
+		append_follower_done(request, 0);
+		return 0;
+	} else if (match != 0) {
+		assert(match == -1);
+		rv = RAFT_SHUTDOWN;
+		goto err_after_request_alloc;
 	}
 
 	/* Delete conflicting entries. */
-	rv = deleteConflictingEntries(r, args, &i);
+	rv = deleteConflictingEntries(r, args, &request->sm, &i);
 	if (rv != 0) {
-		return rv;
+		goto err_after_request_alloc;
 	}
 
 	*rejected = 0;
 
 	n = args->n_entries - i; /* Number of new entries */
+	sm_attr(&request->sm, "n", "%zu", n);
```

Comment:
nit: it's supposed the viewer of attribute graphs will not just into the code right away so it seems it's better to have more wordy and noisy names for attributes. Example:
```
sm_attr(&request->sm, "new_entires_nr", "%zu", n);
```